### DISCUSSION TITLE
Parameter -o for setting outgroup is not working and I'm getting a long branch for the first species (IQTree2.3.4)

### DISCUSSION THREAD

**Discussion by badiaclara** [iqtree/iqtree2#351] [Q&A] [answered]
Hi all,

Thanks in advance for considering helping me out!
I've performed several attempts of a ML reconstruction, but in all of them I'm getting a long branch for the first species in the alignment (I mean REALLY long, and the rest of the tree becomes highly compressed -
![print](https://github.com/user-attachments/assets/4218cb08-fc51-4e1b-9bcb-1924cc0b52ee)
it doesn't make sense biologically speaking).

My attempts follow two steps:

    Running ModelFinder to see what are the best models for my dataset; command:
    iqtree2 -s /storage1/dados/es98149/filoAa/concatAa.phy -p /storage1/dados/es98149/filoAa/charsetAa.nex -m MFP+MERGE -pre concAa -o Spinacia_o

    and then, once having the output "best_scheme.nex" from the first command, I'd run the following command for the phylogeny:
    iqtree2 -s /storage1/dados/es98149/filoAa/concatAa.phy -p /storage1/dados/es98149/filoAa/concatAa.best_scheme.nex -o Spinacia_o -b 500 -pre filoAa-p

In both commands, you'll see that I use the command "-o" to set the outgroup "Spinacia_o", as it is in the .phy files. Please see the chunk from the .phy file below:

(continues...)
Smonacanth .......... .......... .......... .......... .......... .......... .......... .C........
solisioide .......... .......... .........G .......... .......... .......... .......... .C........
Spinacia_o .........A .......... T......... ..A.T..... .......... ....GG.... ..T....... .....G....
Sthurberi .......... .......... .......... .......... .......... .A........ .......... .C........
Sunguispin .......... .......... .......... .......... .......... .......... .......... ..........
(...continues)

I'm aware that in the absence of specifying the outgroup, Iqtree will by default consider the first species as an outgroup. So I tried, as you can see, to set "Spinacia_o" as outgroup, but it does seem to be working.

Any ideas of what can I be doing wrong? Even if you have a really simple idea it might be super helpful for me. I'm clueless here and I've ran over 10 times this phylogeny. Please help a poor PhD trying to defend their thesis! thanks a lot!

**Comment by roblanf** [comment, marked as answer]
@badiaclara can you share the alignment please? This will help in attempting to see what's going on. 

However, a couple of things to point out. First, the outgroup is a purely cosmetic thing for plotting the tree. You can open your tree in any software (FigTree, iTOL, etc) and set a new outgroup to root the tree. The choice of outgroup makes no difference to the tree you are estimating - all trees with reversible models are unrooted anyway. 

Second, that long branch most likely exists because that's the best way to explain the input alignment. It may not make sense biologically, but it probably (I say probably because it's always possible there's a bug!) makes perfect sense with the combination of model and data that you are using. If you can upload your alignment it will be easier to help out more than this.

Rob

**Comment by badiaclara** [comment]
Hi Rob,

Thanks very much for your reply and for the insights. I had a different understanding on how IQtree deals with the outgroup.. I thought it needed to set the outgroup during the analysis to find the best topology. How do I know if my analysis have trees with reversible models?

About the alignment - turns out I found the problem: it was the .phy formatting. I was converting the .nex to .phy on MEGAX, and the first sequence had nucleotides while the others were interpreting the first one as a reference (and MEGAX was giving me dots for equivalent bases instead of keeping the original sequences). So, I converted the .nex to .phy properly in R and it worked beautifully. In addition to that, I removed gap to ensure that I wouldn't have too much data discrepancy. 

Cheers,
Clara

**Reply by roblanf** [reply]
All of the models considered by `-MFP` are reversible, so in the commands above you are using reversible models. 

Great that you fixed up the alignment issues and it's all working now!

**Reply by badiaclara** [reply]
Good to know! Thanks for the contributions!

**Reply by Ziwei-Liu** [reply]
> All of the models considered by `-MFP` are reversible, so in the commands above you are using reversible models.
> 
> Great that you fixed up the alignment issues and it's all working now!

@roblanf  So is it true that if i used -m MFP then i will of course getting an unrooted tree? My command is `iqtree2 -s 04_seqs.aln.ck -m MFP -B 1000 -alrt 1000 -T AUTO --mem 60G -o 'Bmic|XP_combjellyfish'  --prefix 05_seqs_tree` and my tree are like 

`((((Xlae|XP_frogs:0.7008771684,((Hsap|NP_human:0.0732244043,Mmus|NP_Mouse:0.1272135239)100:0.3851232784,Ggal|XP_chicken:0.7830457527)88:0.1547535072)67:0.1549878834,Drer|XP_zebrafish:0.5515779785)87:0.3889550034,(Oedu|XP_oyster:1.0508916239,(Ppal|KAK_tubeworm:1.2622801178,(Lvan|XP_shrimp:0.8234379565,Pcla|XP_crayfish:1.0214783352)100:1.7089963485)42:0.1664327760)39:0.2321136860)65:0.2249689139,Dset|XP_seaurchin:0.7383139339,Bmic|XP_combjellyfish:1.1900600959);`

I used the -o option and find it an unrooted tree, now i think it to be a -m MFP result for an unrooted tree for reversible model.
